lib/vecmat.gi

```diff
@@ -1473,18 +1473,20 @@ local sf, rep, ind, ind2, row, i,big,l;
   fi;
 
   # this can only happen if not big
-  for i in ind do
-    ConvertToVectorRepNC(matrix[i],sf);
-  od;
+  if not big then
```

Comment:
OK, fair enough. I also find the code hard to understand (had to resd it recently foe the ImmutableVector work). I'll takr a other look at this later, and either merge this PR, or do as you suggest and make a new one which rewrites the function with better comments etc.